calico/felix/frules.py

```diff
@@ -409,7 +409,9 @@ def interface_to_chain_suffix(config, iface_name):
     :param iface_name: The interface name
     :returns string: the suffix (shortened if necessary)
     """
-    suffix = iface_name.replace(config.IFACE_PREFIX, "", 1)
-    # The suffix is surely not very long, but make sure.
-    suffix = futils.uniquely_shorten(suffix, 16)
-    return suffix
+    for prefix in config.IFACE_PREFIX:
+        if iface_name.startswith(prefix):
+            iface_name = iface_name[len(prefix):]
+            break
```

Comment:
Done. This is probably not the most optimal solution in terms of performance, but since there won't be many prefixes it might be acceptable. As an alternative IFACE_PREFIX can be sorted in `_validate_cfg`, but it will bring a side-effect to `interface_to_chain_suffix` and will affect code supportability.